### PR TITLE
For `state run` or `state exec`, only interpret "-v" as verbose if it's the first arg.

### PR DESCRIPTION
**Pull request by mitchell-as**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-928" title="DX-928" target="_blank">BugDX-928</a>  argument to scripts -v is interpreted by the state tool
  </td></table>
  <br />

### CODE REVIEW

**Comment by daved** [comment]
@mitchell-as Please consider not setting the "Persistent" field to true and dropping as much related logic as possible. If this is still the best approach after that consideration, I'll approve this PR as soon as I read your reply.

**Comment by daved** [comment]
Ah, sorry. I'm suggesting that, instead of manipulating the args themselves, we change the rules of the flags. This has to do with how our captain/cobra library works. The relevant "Persistent" field is set here: https://github.com/ActiveState/cli/blob/1c4b8865440a3f146c90fb13f920991f15fbcf17/cmd/state/internal/cmdtree/cmdtree.go#L225-L236

If "verbose" is no longer set, it should only be valid for the top-level command and no longer honored for subcommands. But I leave that to you to decide, but @Naatan may want to weigh in on this too.

**Comment by Naatan** [comment]
This is purposefully not using the captain flag registration as that comes into play too late in our runtime. We would be missing out on verbose logs if we waited for the proper flag registration.

**Comment by daved** [comment]
Understood. That makes me think changes need to happen both places. IOW: Detect early and only for top-level, don't manipulate the args, turn off Persist.

**Comment by mitchell-as** [comment]
So can this PR proceed as-is? Or do I need to do more?

**Comment by Naatan** [comment]
I don't understand why you want to change Persist? `-v` is still a global flag. That's not changing.

**Comment by daved** [comment]
It makes more sense to me that we alter the nature of the verbose flag rather than post process the arguments.

File: cmd/state/main.go
Comment (daved) [inline cmd/state/main.go:265]: This will detect the run or exec commands only when they are the first argument. If another flag is set as the first argument, then the command will not be detected. This increases my thought that this issue should be resolved by altering the flag handling rather than post-processing the args.

**Review by daved — CHANGES_REQUESTED** [review]

**Comment by mitchell-as** [comment]
@daved ping
As noted, we have to look for verbose flags ASAP, so modifying the existing cmdtree handlers isn't early enough. Also, they'll find the verbose flag anyway for `state run` and `state exec` commands, no matter where they are. We only want to run in verbose mode if the flag comes immediately after the command, rather than anywhere in that command's argument list.

**Comment by mitchell-as** [comment]
Okay, thanks for the clarification. I've turned off "Persist" for the verbose flag as you requested.

File: cmd/state/main.go
Comment (daved) [inline cmd/state/main.go:272]: I've not reasoned this concern fully, but... I think this will trip up on `state --output json`. I wonder if the spf13/cobra or spf13/pflag packages have helper functions we can leverage.

**Review by daved — CHANGES_REQUESTED** [review]

File: cmd/state/internal/cmdtree/exec.go
Comment (mitchell-as) [reply cmd/state/internal/cmdtree/exec.go:39]: Looks like it's necessary. If I don't skip over '-v', I get an error `The script '-v' is not defined in activestate.yaml.`

File: cmd/state/main.go
Comment (mitchell-as) [reply cmd/state/main.go]: You are right. I couldn't find any helper functions. If there were, there'd be a lot of repetition thanks to our `cmdtree.New()` structure. Perhaps this is one of those cases where "good enough" applies, rather than 100% correct.

**Comment by mitchell-as** [comment]
I put the `Persist` field back in because commands like `state config set -v foo bar` no longer turn on verbose logging, and I would expect them to do so. Otherwise we require `state -v config set foo bar` or `state config -v set foo bar`, which seems a bit unreasonable to me.

File: cmd/state/internal/cmdtree/exec.go
Comment (daved) [reply cmd/state/internal/cmdtree/exec.go:39]: Let's go with it for now, with a bug I've filed: https://activestatef.atlassian.net/browse/DX-960

File: cmd/state/main.go
Comment (daved) [reply cmd/state/main.go]: I hear you, but I disagree that this is good enough. Unless we document that double dash flags cannot use spaces and must use the equals sign. But that breaks POSIX, I think (I've not looked it up, but it seems like it would).

Comment (daved) [reply cmd/state/main.go]: As for taking a direction, please go wit this for now and file a bug for us to resolve handling this issue. It may be related to the other bug I've filed, but I suggest we decide that during triage.

Comment (mitchell-as) [reply cmd/state/main.go]: Sorry, what I mean by "good enough" was my most recent commit: https://github.com/ActiveState/cli/pull/1860/commits/6f9b08ce0f041aa3b31f5e4957f84cf7c469a76b

It does not trip up on `--option value` flags.

Comment (mitchell-as) [reply cmd/state/main.go]: I've filed a bug: https://activestatef.atlassian.net/browse/DX-961

**Review by daved — APPROVED** [review]

**Pull request merged**
